docs/reference/notion.md

```diff
@@ -0,0 +1,113 @@
+# Notion Connector
+
+The [Elastic Notion connector](../connectors/sources/notion.py) is built with the Elastic connectors Python framework and is available as a self-managed [connector client](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
+
+## Availability and prerequisites
+
+This connector is available as a **connector client** from the **Python connectors framework**. To use this connector, satisfy all [connector client requirements](https://www.elastic.co/guide/en/enterprise-search/master/build-connector.html).
+
+This connector is in **beta** and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.
+
+## Usage
+
+To use this connector as a **connector client**, select the **Notion** tile when creating a new connector under **Search -> Connectors**.
+
+For additional operations, see [Usage](https://www.elastic.co/guide/en/enterprise-search/master/connectors-usage.html).
+
+## Connecting to Notion
+
+To connect to Notion, the user needs to [create an internal integration](https://www.notion.so/help/create-integrations-with-the-notion-api#create-an-internal-integration) for their Notion workspace, which can access resources using the Internal Integration Secret Token. Configure the Integration with following settings:
+
+1. Users must grant `READ` permission for content, comment and user capabilities for that integration from the Capabilities tab.
+
+2. Users must manually [add the integration as a connection](https://www.notion.so/help/add-and-manage-connections-with-the-api#add-connections-to-pages) to each page and database that needs to be retrieved.
```

Comment:
This is aligned with the Notion documentation, and seems to be one of their built in limitations (along with no timestamps returned and no DLS support). It would be great if inheriting is an option - @praveen-elastic could you please check whether Sean's suggestion is feasible?

Reply:
I checked and found that sub-pages are able to inherit the connections of parent page regardless of whether the sub-page was present before adding the connection or the sub-page is created after the connection is added to the parent page.

Reply:
I've updated the statement specifying that the connection needs to be added only in the top-level pages